pyincore/analyses/mlenabledcgeslc/mlcgeslc.py

```diff
@@ -0,0 +1,197 @@
+from typing import Tuple, List, Dict
+
+import numpy as np
+import os
+import pandas as pd
+
+from pyincore import globals as pyglobals
+from pyincore.client import IncoreClient
+from pyincore.analyses.core_cge_ml import CoreCGEML
+from pyincore.utils import parse_files
+
+logger = pyglobals.LOGGER
+
+
+class MlEnabledCgeSlc(CoreCGEML):
+
+    model = "Machine Learning Enabled Computable General Equilibrium - Salt Lake City "
+
+    #### Coefficients files
+    DDS_coefficients_file = "DDS_coefficients_n.csv"
+    DY_coefficients_file = "DY_coefficients_n.csv"
+    MIGT_coefficients_file = "MIGT_coefficients_n.csv"
+    DFFD_coefficients_file = "DFFD_coefficients_n.csv"
+    # DDS_coefficients_file = "DDS_coefficients.csv"
+    # DY_coefficients_file = "DY_coefficients.csv"
+    # MIGT_coefficients_file = "MIGT_coefficients.csv"
+    # DFFD_coefficients_file = "DFFD_coefficients.csv"
+
+    #### Base value files
+    DS_base_val_file = "DS_base_val.csv"
+    GI_base_val_file = "GI_base_val.csv"
+    HH_base_val_file = "HH_base_val.csv"
+    FD_base_val_file = "FD_base_val.csv"
+
+    Base_KAP_file = "baseKAP.csv"
+
+    base_file_path = os.path.join(
+        pyglobals.PYINCORE_PACKAGE_HOME,
+        "analyses",
+        "mlenabledcgeslc",
+    )
+
+    model_filenames = {
+        "ds": os.path.join(
+            base_file_path,
+            DDS_coefficients_file,
+        ),
+        "dy": os.path.join(
+            base_file_path,
+            DY_coefficients_file,
+        ),
+        "migt": os.path.join(
+            base_file_path,
+            MIGT_coefficients_file,
+        ),
+        "dffd": os.path.join(
+            base_file_path,
+            DFFD_coefficients_file,
+        ),
+    }
+
+    filenames = [
+        os.path.join(
+            base_file_path,
+            DS_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            GI_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            HH_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            FD_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            Base_KAP_file,
+        ),
+    ]
+
+    def __init__(self, incore_client: IncoreClient):
+        sectors, base_cap_factors, base_cap, model_coeffs, cap_shock_sectors = parse_files(self.model_filenames, self.filenames)
+        self.base_cap_factors = base_cap_factors
+        self.base_cap = base_cap
+        self.model_coeffs = model_coeffs
+        self.cap_shock_sectors = cap_shock_sectors
+        super(MlEnabledCgeSlc, self).__init__(incore_client, sectors, labor_groups=[f"L{gp}" for gp in range(1, 5)]) # 4 labor groups
+
+    def get_spec(self):
+        return {
+            'name': 'Salt-Lake-cge',
+            'description': 'CGE model for Salt Lake City.',
+            'input_parameters': [
+                {
+                    "id": "domestic_supply_fname",
```

Comment:
This seems overly complicated. Why not just follow the pattern from the other analyses and set the resultName and then the output will be something like:
result_name + household-count.csv. So if a user sets nothing, then the output is just household-count.csv and if they set "SLC" then it's SLC_household-count.csv? It seems like a lot of extra parameters to allow control over each filename.

Also, this poses a problem for running this is a workflow if the output file can be anything. The workflow uses wildcards to find output files so it's better if there is a fixed portion of the name. Then household count can be find by telling the workflow to look for a file called:

"*household-count.csv"

Reply:
I see, I thought we wanted to have control over individual file names, I can change it to the resultName if that is required.

Reply:
@longshuicy What are your thoughts? I think my preference is to match other analyses and add a setResultName that is then used to name each output. If we want to let users set the names, that's fine, but then we need to make sure there is a static part to the name for the reasons above with finding output files.

Reply:
So, for our flow, we can just run the analysis as is which generates the original names as found in old CGE. But nevertheless, I have removed the feature allowing users to name each file, and just added result_name input param which is optional unlike other analyses. This input is a prefix to the original file names so the filenames will still have the original names before file extension.

Reply:
Sorry late to the game. 

> and just added result_name input param which is optional unlike other analyses. 

This sounds good to me.